ouroboros-network-protocols/testlib/Ouroboros/Network/Protocol/Handshake/Test.hs

```diff
@@ -971,6 +983,63 @@ prop_query_version createChannels codec versionDataCodec clientVersions serverVe
     clientVersions' = setQueryVersions clientVersions
 
 
+-- | Run a query for the server's supported version.
+--
+prop_peerSharing_symmetric :: ( MonadAsync m
+                           , MonadCatch m
+                           , MonadST m
+                           )
+                           => m (Channel m ByteString, Channel m ByteString)
+                           -> Codec (Handshake NodeToNodeVersion CBOR.Term)
+                                     CBOR.DeserialiseFailure m ByteString
+                           -> VersionDataCodec CBOR.Term NodeToNodeVersion ArbitraryNodeToNodeVersionData
+                           -> Versions NodeToNodeVersion ArbitraryNodeToNodeVersionData Bool
+                           -> Versions NodeToNodeVersion ArbitraryNodeToNodeVersionData Bool
+                           -> m Property
+prop_peerSharing_symmetric createChannels codec versionDataCodec clientVersions serverVersions = do
+  (clientRes, serverRes) <-
+    runConnectedPeers
+      createChannels nullTracer codec
+      (handshakeClientPeer
+        versionDataCodec
+        acceptableVersion
+        clientVersions)
+      (handshakeServerPeer
+        versionDataCodec
+        acceptableVersion
+        queryVersion
+        serverVersions)
+  pure $ case (clientRes, serverRes) of
+    (  Right (HandshakeNegotiationResult _ v (ArbitraryNodeToNodeVersionData clientResult))
+     , Right (HandshakeNegotiationResult _ v' (ArbitraryNodeToNodeVersionData serverResult))
+     ) | v == v'
```

Comment:
it's a bit strange that `runConnectedPeers` returns a pair of `ArbitraryNodeToNodeVersionData` rather than a pair of `NodeToNodeVersionData`.

Reply:
Yes, I was just lazy not to massage the data types, `ArbitraryNodeToNodeVersionData` has an Acceptable instance so it just works

Reply:
If you don't want to do that now, please leave a TODO / issue to refactor this later.